### PR TITLE
wails : add first screen for define password

### PR DESCRIPTION
**Pull request by maxime-carabina**
We are adding the screen to define password on wails.

### CODE REVIEW

File: wails-frontend/src/layouts/DefaultLayout/DefaultLayout.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { ReactNode } from 'react';
+
+interface LayoutProps {
+  children: ReactNode;
+}
+
+export default function DefaultLayout(props: LayoutProps) {
+  return <div className={`bg-primary min-h-screen`}>{props.children}</div>;
```
Comment (pivilartisant) [inline wails-frontend/src/layouts/DefaultLayout/DefaultLayout.tsx:8]: I think we should make a separate pr for the default layout and create and refactor the wails screen to implement

File: wails-frontend/src/pages/passwordPrompt.tsx [hunk 2]
```diff
@@ -68,50 +72,61 @@ const PasswordPrompt = () => {
...
-      <div>
-        <button className="btn" onClick={handleCancel}>
-          Cancel
-        </button>
-        <button className="btn" onClick={applyPassword}>
-          {applyStr(req)}
-        </button>
+        {(req.Action === promptAction.newPasswordReq ||
+          req.Action === promptAction.importReq) && (
+          <div id="input" className="pt-4">
```
Comment (pivilartisant) [inline wails-frontend/src/pages/passwordPrompt.tsx:99]: Bis répétita see above

**Review by pivilartisant — CHANGES_REQUESTED** [review]

File: wails-frontend/src/pages/passwordPrompt.tsx [hunk 3]
```diff
@@ -68,50 +72,61 @@ const PasswordPrompt = () => {
...
-        <div id="input" className="input-box">
-          <input
-            id="name"
-            className="input"
-            onChange={updatePasswordConfirm}
-            autoComplete="off"
+    <DefaultLayout>
+      <div
+        className={`flex flex-col justify-center h-screen
+        max-w-xs min-w-fit text-f-primary m-auto`}
```
[the remark this answers is not in the document]
Comment (maxime-carabina) [reply wails-frontend/src/pages/passwordPrompt.tsx:85]: Done

File: wails-frontend/src/pages/passwordPrompt.tsx [hunk 4]
```diff
@@ -68,50 +72,61 @@ const PasswordPrompt = () => {
...
+      >
+        <h1 className="mas-title">{req.Msg}</h1>
+        <p className="mas-body pt-4">{baselineStr(req)}</p>
+        <div id="input" className="pt-4">
```
[the remark this answers is not in the document]
Comment (maxime-carabina) [reply wails-frontend/src/pages/passwordPrompt.tsx:89]: I missed to remove the id. It's useless

File: wails-frontend/src/pages/passwordPrompt.tsx (hunk 2 above)
Comment (maxime-carabina) [reply wails-frontend/src/pages/passwordPrompt.tsx:99]: Same

**Review by pivilartisant — APPROVED** [review]

File: wails-frontend/src/pages/passwordPrompt.tsx [hunk 5]
```diff
@@ -66,52 +69,64 @@ const PasswordPrompt = () => {
...
+            </Button>
+          </div>
+          <Button
+            preIcon={<FiLock />}
+            onClick={applyPassword}
+            disabled={
+              req.Action ===
+              (promptAction.newPasswordReq || promptAction.importReq)
+                ? !(password && passwordConfirm)
+                : !password
```
Comment (peterjah) [inline wails-frontend/src/pages/passwordPrompt.tsx:122]: Are you sure its working ??? 
imho
```
req.Action ===
    (promptAction.newPasswordReq || promptAction.importReq)
 ```
will always return false
maybe you can create a new function to factorize this:

const needConfirm = (req: promptRequest) => 
req.Action === promptAction.newPasswordReq  || req.Action === promptAction.importReq

File: wails-frontend/src/pages/passwordPrompt.tsx [hunk 6]
```diff
@@ -9,24 +9,27 @@ import { events, promptAction, promptRequest } from '../events/events';
...
+import { Password, Button } from '@massalabs/react-ui-kit';
+
 const PasswordPrompt = () => {
   const nav = useNavigate();
 
   const { state } = useLocation();
   const req: promptRequest = state.req;
 
-  const applyStr = (req: promptRequest) => {
+  function applyStr(req: promptRequest) {
```
Comment (peterjah) [inline wails-frontend/src/pages/passwordPrompt.tsx:21]: you don't like arrow functions ?
Or is there a decision made to explicitely not use it ?

**Review by peterjah — CHANGES_REQUESTED** [review]

File: wails-frontend/src/pages/passwordPrompt.tsx (hunk 5 above)
Comment (maxime-carabina) [reply wails-frontend/src/pages/passwordPrompt.tsx:122]: @peterjah I tested it in front and it works!

image

image

image

File: wails-frontend/src/pages/passwordPrompt.tsx (hunk 6 above)
Comment (maxime-carabina) [reply wails-frontend/src/pages/passwordPrompt.tsx:21]: @peterjah We decided in the frontend team to replace const by function.
For example :
- onClick
- onChange
- etc...

But with values like `useState` or simple attribution like `var test = "this is a test";`, we use const.

File: wails-frontend/src/pages/passwordPrompt.tsx (hunk 5 above)
Comment (maxime-carabina) [reply wails-frontend/src/pages/passwordPrompt.tsx:122]: in Logic :
`req.Action === (promptAction.newPasswordReq || promptAction.importReq)`

AND

``(req.Action === promptAction.newPasswordReq) || (req.Action === promptAction.importReq)

File: wails-frontend/src/pages/passwordPrompt.tsx [hunk 7]
```diff
@@ -66,52 +69,64 @@ const PasswordPrompt = () => {
...
-      )}
-      <div>
-        <button className="btn" onClick={handleCancel}>
-          Cancel
-        </button>
-        <button className="btn" onClick={applyPassword}>
-          {applyStr(req)}
-        </button>
+        {(req.Action === promptAction.newPasswordReq ||
+          req.Action === promptAction.importReq) && (
```
Comment (mazmassa) [inline wails-frontend/src/pages/passwordPrompt.tsx:98]: Upper in code:

```suggestion
        {
        let { newPasswordReq, importReq } = promptAction;
        let isNewPassword = req.Action === newPasswordReq;
        let isImport = req.Action === importReq;

        {(isNewPassword || isImport) && (
```

```suggestion
        {(req.Action === promptAction.newPasswordReq ||
          req.Action === promptAction.importReq) && (
```

File: wails-frontend/src/pages/passwordPrompt.tsx (hunk 5 above)
Comment (mazmassa) [reply wails-frontend/src/pages/passwordPrompt.tsx:122]: Even this is working is a nightmare to understand. Let's extract that logic and make things easy to understand... we are mixing OR and ! just hell to follow...

File: wails-frontend/src/pages/passwordPrompt.tsx (hunk 6 above)
Comment (maxime-carabina) [reply wails-frontend/src/pages/passwordPrompt.tsx:25]: Done

File: wails-frontend/src/pages/passwordPrompt.tsx (hunk 7 above)
Comment (maxime-carabina) [reply wails-frontend/src/pages/passwordPrompt.tsx]: Done

File: wails-frontend/src/pages/passwordPrompt.tsx [hunk 8]
```diff
@@ -66,52 +73,61 @@ const PasswordPrompt = () => {
...
+        <div className="pt-4 flex gap-4">
+          <div className="max-w-min">
+            <Button variant={'secondary'} onClick={handleCancel}>
+              Cancel
+            </Button>
+          </div>
+          <Button
+            preIcon={<FiLock />}
+            onClick={applyPassword}
+            disabled={isDisabled}
```
Comment (mazmassa) [inline wails-frontend/src/pages/passwordPrompt.tsx:124]: 👏

File: wails-frontend/src/pages/passwordPrompt.tsx [hunk 9]
```diff
@@ -66,52 +73,61 @@ const PasswordPrompt = () => {
...
+          >
+            {applyStr(req)}
```
[the remark this answers is not in the document]
Comment (maxime-carabina) [reply wails-frontend/src/pages/passwordPrompt.tsx:126]: @mazmassa I kept the name that Peter had put. I will change it to `textButton` if you prefer.

[the remark this answers is not in the document]
Comment (mazmassa) [reply wails-frontend/src/pages/passwordPrompt.tsx:126]: If that is just to get a content label for the button, why not: 

**buttonLabel** or something related? 🤔 and create a getter there:
```js
 get buttonLabel() {
    return ...
  }
```

[the remark this answers is not in the document]
Comment (maxime-carabina) [reply wails-frontend/src/pages/passwordPrompt.tsx:126]: > If that is just to get a content label for the button, why not:
> 
> **buttonLabel** or something related? thinking and create a getter there:
> 
> ```js
>  get buttonLabel() {
>     return ...
>   }
> ```

Oh right! It's better with this name.

File: wails-frontend/src/pages/passwordPrompt.tsx [hunk 10]
```diff
@@ -9,43 +9,50 @@ import { events, promptAction, promptRequest } from '../events/events';
 import { useLocation, useNavigate } from 'react-router-dom';
 import { handleApplyResult, handleCancel } from '../utils/utils';
 
+import { FiLock } from 'react-icons/fi';
+import { Password, Button } from '@massalabs/react-ui-kit';
+
 const PasswordPrompt = () => {
   const nav = useNavigate();
```
Comment (mazmassa) [inline wails-frontend/src/pages/passwordPrompt.tsx:16]:
```suggestion
  const navigate = useNavigate();
```

**Review by peterjah — APPROVED** [review]

File: wails-frontend/src/pages/passwordPrompt.tsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (maxime-carabina) [reply wails-frontend/src/pages/passwordPrompt.tsx:126]: @mazmassa I will change `baselineStr` function name too because it's not explicit.
We can try `instructionStr` ?

File: wails-frontend/src/pages/passwordPrompt.tsx (hunk 10 above)
Comment (maxime-carabina) [reply wails-frontend/src/pages/passwordPrompt.tsx:16]: Done

File: wails-frontend/src/pages/passwordPrompt.tsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (mazmassa) [reply wails-frontend/src/pages/passwordPrompt.tsx:126]: Why str!? what is instruction or baseline!? 🤔 

buttonLabel<Something>...

[the remark this answers is not in the document]
Comment (maxime-carabina) [reply wails-frontend/src/pages/passwordPrompt.tsx:126]: image

No this is the subtitle here

**Review by mazmassa — APPROVED** [review]
We fixed in a call. LGTM!

**Review by mazmassa — APPROVED** [review]
We fixed in a call. LGTM!

File: wails-frontend/src/pages/passwordPrompt.tsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (maxime-carabina) [reply wails-frontend/src/pages/passwordPrompt.tsx]: Done
